src/main/java/ch/njol/skript/expressions/ExprEnchantmentLevel.java

```diff
@@ -102,42 +92,52 @@ public boolean isSingle() {
 				return null;
 		}
 	}
-	
+
 	@Override
 	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
-		if (delta == null)
-			return;
-		
-		ItemType[] source = items.getArray(e);
+		ItemType[] itemTypes = items.getArray(e);
 		Enchantment[] enchantments = enchants.getArray(e);
-		int newLevel = ((Number) delta[0]).intValue();
-		
-		for (ItemType item : source) {
-			if (!item.hasAnyEnchantments(enchantments))
-				continue;
-			
-			EnchantmentType[] enchants = item.getEnchantmentTypes();
-			assert enchants != null; // Can't be null at this point due to the above check
-			for (EnchantmentType enchant : enchants) {
-				item.removeEnchantments(enchant);
-				Enchantment type = enchant.getType();
-				int changed = newLevel;
-				assert type != null;
-				
-				if (mode == ChangeMode.ADD)
-					changed = Math.max(0, enchant.getLevel() + changed);
-				else if (mode == ChangeMode.REMOVE)
-					changed = Math.max(0, enchant.getLevel() - changed);
-				
-				if (changed > 0)
-					item.addEnchantments(new EnchantmentType(type, changed));
+		int changeValue = ((Number) delta[0]).intValue();
+
+		for (ItemType itemType : itemTypes) {
+			for (Enchantment enchantment : enchantments) {
+				EnchantmentType enchantmentType = itemType.getEnchantmentType(enchantment);
+				int oldLevel = enchantmentType == null ? 0 : enchantmentType.getLevel();
+
+				int newItemLevel;
+				switch (mode) {
+					case ADD:
+						newItemLevel = oldLevel + changeValue;
+						break;
+					case REMOVE:
+						newItemLevel = oldLevel - changeValue;
+						break;
+					default: // case SET
```

Comment:
Do you think it would be better to directly specify SET as a case and then default can just assert false or something?

Reply:
Yea, sure